### ISSUE TITLE
floating dock widgets get into unclosable state

### ISSUE THREAD

**Issue by GitMensch** [KDAB/hotspot#467]
The main reason for this question: I've moved out the disassembly view to increase it - and for some reason it does not respond at all afterwards.
(Side question: Can anyone reproduce that?)

So: How to reset the disassembly view (or all views)?

**Comment by milianw** [comment]
what do you mean exactly, I don't follow - can you show a screenshot and add more context to better describe the issue you are having?

**Comment by GitMensch** [comment]
This is what Hotspot currently looks like on this machine:

![sample](https://user-images.githubusercontent.com/6699539/221432300-99d3a075-b76b-4b4b-96a9-e6f76fabdcf2.png)

I can click the two views that were moved to a separate windows and get them the focus, but neither resizing nor moving, nor close works - and I can't scroll anymore in them.

Therefore - separate from the underlying problem - I'd like to reset all views so they are "contained" in the main window again.
I've looked for a "reset views/windows" command in the View or Windows menu but did not found any.

**Comment by milianw** [comment]
we have no such option yet, might be a good idea to bring it back. and to double check - clicking on the `x` buttons doesn't do anything? what happens if you double-click on the titles, that should dock the floating dock widget again?

generally though that config is stored somewhere in `~/.config/hotspotrc` I believe, try to nuke that file and see if it fixes the immediate issue.

then explain how you got into that state in the first place and we can try to reproduce and fix the issue(s)

**Comment by GitMensch** [comment]
Neither double-click on the title nor the close button works at all.
Deleting `~/.config/hotspotrc` did the trick!

Then just moving any view out of the window by click+drag gets me back to this state on this machine :-/ but as deleting the config works here...

The original question is answered, should we close it or change the title and track a feature-request for a menu option to reset the views/windows here?

**Comment by milianw** [comment]
let's keep it here, no need for a new task

**Comment by GitMensch** [comment]
It is Wayland via X11-Server redirected... It seems that nearly everything works, but the [KDDockWidgets](https://github.com/KDAB/KDDockWidgets).
KDE in general works fine, at least when checking the "normal" windows (settings, file open/save, ... in Hotspot and KCachegind - _possibly_ the only other KDE based app used on that machine - in any place).

**Comment by milianw** [comment]
what does "wayland via x11-server redirected" mean? xwayland? what display manager?

**Comment by GitMensch** [comment]
Correction: if I set `QT_QPA_PLATFORM=banana I get an error message telling me only xcb is available. If I do that the same and call kcachgegrind I get `Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.`

... so: Is this an issue with the appimage? Could there be added support for more platform plugins?

**Comment by milianw** [comment]
yes, could be that this is the issue, unsure when I'll find the time to investigate this
